nfsv4/Makefile

```diff
@@ -2,18 +2,30 @@
 
 all: nfs4.so
 
-OBJ = rpc.o nfs4.o
+OBJ = rpc.o xdr.o client.o
+SQLITE_OBJ = nfs4.o $(OBJ)
 
-.c.o: buffer.h nfs.h
-	gcc -g -fPIC -I. -std=gnu99 -I$(SQLITE) $< -c
+# set -DTRACE on nfs4.c for operation printfs
+#TRACING = -DTRACE
 
-nfs4.so: check-env $(OBJ)
-	ld -shared $(OBJ) -o nfs4.so
+nfs4.o: nfs4.c
+ifndef SQLITE
+	echo 'SQLITE is undefined'
+else
+	gcc -g -fPIC -I. $(TRACING) -std=gnu99 -I$(SQLITE) $< -c
+endif
+
+# how to depend on buffer.h nfs.h 
```

Comment:
I can't answer this because I don't know make well. I'm more inclined to invest in learning cmake so perhaps we can plan on switching at some point.